cpp/mg_utility/mg_graph.hpp

```diff
@@ -279,6 +333,9 @@ class Graph : public GraphView<TSize> {
 
   std::vector<std::vector<TSize>> adj_list_;
   std::vector<std::vector<TNeighbour>> neighbours_;
```

Comment:
Do neighbours_ contains both in and out neighbours or just out? If so make it explicit

Reply:
Due to the way `neighbours_` is initialized, with directed graphs it only contains the out-neighbors.
Keeping in mind that in- and out-neighbors are identical for undirected graphs, I’m reluctant to rename the vector, but I’ll clarify this with a comment. Does this sound good to you?

Reply:
Yes, that sound ok for now. But I would definitely think about separating different versions of Graph, since now you have directed/undirected/weighted in the same class, and that causes undirected to have properties it doesn't use. What do you think about this @jmatak ?

Reply:
If we decide to separate them, undirected graphs would still have to implement certain interfaces because of generality. For example, if an algorithm requires the out-neighbors, we should be able to call `OutNeighbours` without having to check whether the graph is directed. Of course, for undirected graphs, `OutNeighbours` would just default to calling `Neighbours`. 

Does the above make sense? I believe we should definitely take it into account if it’s a valid suggestion.

Reply:
aha, so we are speaking in the context of algorithms running on both undirected and undirected graphs (sorry for the late response)?

Reply:
Right @jbajic!